src/charm.py

```diff
@@ -47,6 +52,67 @@
 logger = logging.getLogger(__name__)
 
 
+def monotonize(value, default_value, multiplier):
+    """A continuous function from the given value to a multiplier-adjusted value.
+
+    >>> monotonize(1000, 200, 0.8)  # take 0.8 * 1000
+    800.0
+    >>> monotonize(260, 200, 0.8)  # take 0.8 * 260 (input value still > default / 0.8)
+    208.0
+    >>> monotonize(250, 200, 0.8)  # take the default, 200 (input value <= default / 0.8)
+    200
+    >>> monotonize(200, 200, 0.8)
+    200
+    >>> monotonize(150, 200, 0.8)  # return the input value (input value < default)
+    150
+    """
+    if value is None:
+        return default_value
+
+    adjusted_value = value * multiplier
+    if adjusted_value > default_value:
+        return adjusted_value
+
+    if value > default_value:
+        return default_value
+
+    return value
+
+
+def convert_k8s_quantity_to_legacy_binary_gigabytes(
```

Comment:
Could this be a utility methods in the Kubernetes resource patch library ? Seems like it is useful primarily in that context.

Reply:
Thanks. Though I am not aware myself, it may be worth considering if there are other applications which may want to use the Kubernetes resource patch library that do similar things like Prometheus in this regard